Admin Tech: In-Game Tools, Outfits, and IC Spawning System Content Bundle

ImogenOC opened the pull request:
## About The Pull Request
_I never claimed to be good at this, but that won't stop me. I'm a woman who knows enough to be dangerous._
The bluespace technician has an absolutely atrocious loadout. Dated kit selections, lacking a gas analyzer, missing a brpd, uses a combat outfit instead of an admin suit, just an incredible amount level of oversight in it's design. You might not spend any time in this suit, but I spend an ungodly amount inside of one. I'm tired of spawning items that I need frequently, var-editing items to access their unique functions, or generally dealing with tools that were designed for IC usage that don't have a debug or admin variation. Like, did you know the BST suit doesn't have built in anti-slip? It is embarrassing to watch your god-character eat shit over an oil spill, or some water vapor. 

This PR started small and slowly possessed me. It's a common story. I have come from itching my head trying to add singular items, to adapting complex systems and working with client and system prefs. I have spent hundreds of hours doing research on different systems, the structure of our codebase, learning how dm sticks itself together, and just oodles more. I can actually consider myself to be workable for handling our code now. While I don't find myself in a position to make completely original stuff, I've learned how much of our pre-existing codebase works. All of this over a goddamn gas analyzer. Not only have I come a long ways with my dev skills, but I've grown in how I see and want to interact with Nova. This PR has even spun off a few other fixes and content additions that I could not have done if not for the exploration I did in here.

In it's end state, this PR is a massive bolt on of what I perceive to be missing content in both testing, event, and educational tools. I had so many 'wait, this exists?' moments making this PR as I explored and dug through our codebase. By grouping everything useful and interesting together in linked systems and outfits, interacting with this module should help to streamline and raise the general game literacy and competency of Staff and anyone working on local servers.

Here's just a smattering of what I think are the most interesting additions:
* Completely brand new Admin Tech outfits with special functions such as quick phasing, quick space walking, quick godmode toggles, phasing and infinite reach, integrated stable extract management systems, trait groups to help stop the world around you from hating you, and plenty of other small considerations
* Modernized Ctrl + Click IC Spawning system, with SAVE SLOTS.
* Debug Boxes spawning system
* Updated tools, with attempted consideration on organizations
* A new holosign management system
* The dagenblicky, a special pen hidden somewhere in the PR contents
* New keybinds and verbs for wallhacks and combohud and stuff
* A new techweb
* Cellguns can use a radial now
* New borgs with unbreakable polymorphs
* Global fixes to the width of storage columns
* ID CARDS THAT ACTUALLY HAVE GLOBAL ACCESS ON OUR DATABASE

[You know that one scene from malcom in the middle](https://www.youtube.com/watch?v=AbSehcT19u0)? 
That's me.

Here is some general trivia about this monstrosity, which will probably only grow with time:
Over 140 new items (/obj/item/...) 
45 datums
73 files touched

You should struggle for nearly nothing ICly as a staffer with this PR. If you have an idea, let me know.

**Credits:**
ERIS - Sprites and Sprites for Reference.

## How This Contributes To The Nova Sector Roleplay Experience
Admin techs fixing things are cooler than the hand of god warping reality around them.
Admins that are having more fun and hate the game less, are less likely to hate their players.

## Proof of Testing
Will populate once everything is in place.
<details>
<summary>Screenshots/Videos</summary>
  
</details>

## Changelog
:cl: Moonridden, see readme.md
refactor: Updates the Observer Ctrl + Click IC Spawning System to 'Modern' standards with tguilists, a new silent spawn method, and a kinda funky save-slot system which reduces clicks-to-spawn down to two.
admin: Over 140 new Bluespace and 'Subspace' themed items, covering everything from event tools, specialized clothing with unique functions, updates of old debug and administrative tools, to fun objects.
admin: New Bluespace and Subspace Technician outfits, custom tailored to provide you as many IC solutions to problems as possible
admin: New Admin cyborgs, and new tachikoma models, thank you ERIS you are beautiful
admin: New Debug Box with a list based item population system.
admin: Adds a new wallhacks verb, updates combohud, adds keybind options for both
code: New checks for widescreen settings on storage ui width expansion
code: New privacy sign management system and verb, respecting user preferences
refactor: Cellguns, like the Veymed, now have a special click interact to choose between processional interacts for cell selection, or a radial, saving per gun.
map: Nova specific plushie boxes and spawners
image: Lots of new codersprite icons.

/:cl:

:cl:
add: Added new mechanics or gameplay changes
add: Added more things
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
sound: added/modified/removed audio or sound effects
image: added/modified/removed some icons or images
map: added/modified/removed map content
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

ImogenOC committed:
bluespace tech rework initial

this should probably be a module at this point
a new modsuit, cleans up old stuff, makes new loadout adjustments, improves drip, adds missing debug items from to outfit, creates debug tools, adds new and frankly ugly booties

ImogenOC committed:
makes the subspace modsuit its own thing

ImogenOC committed:
more pr work

new assets, new tools, new organization, more detail work

ImogenOC committed:
Merge branch 'NovaSector:master' into bluespace-tech-rework

ImogenOC committed:
make admin_tech module

guts icspawning module for spare parts, integrates into new module
breaks up my cancer file into separate files
I bet someone's gonna make me use
folders

ImogenOC committed:
Create skillchip.dm

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into bluespace-tech-rework

ImogenOC committed:
more fixes and tunings

more organization
addresses a runtime with the bsrped
new skin suit for techs
fixes the new holochip
tries to address an issue with new goggles, fails.
todo listings

ImogenOC committed:
new berets and flannels for the two new outfits

also reorgs observer spawn menu
fixes pinpointer bool

ImogenOC committed:
new gasmask, filters, backlinks to item files for reference

ImogenOC committed:
Merge branch 'NovaSector:master' into bluespace-tech-rework

ImogenOC committed:
attempts to unfuck .dme again

Zergspower commented on modular_nova/modules/admin_tech/code/admin_items.dm:
```diff
@@ -0,0 +1,276 @@
+		/obj/item/stack/rods/shuttle = 50,
+		/obj/item/stack/sheet/glass/fifty = null,
+		/obj/item/stack/sheet/rglass/fifty = null,
+		/obj/item/stack/sheet/mineral/plasma = 50,
```
```suggestion
		/obj/item/stack/sheet/mineral/plasma/fifty = null,
```

ImogenOC committed:
successfully unfucks the dme.

Zergspower commented on modular_nova/modules/admin_tech/code/admin_items.dm:
```diff
@@ -0,0 +1,276 @@
+		/obj/item/stack/sheet/plastic/fifty = null,
+		/obj/item/stack/sheet/runed_metal/fifty = null,
+		/obj/item/stack/sheet/mineral/abductor = 50,
+		/obj/item/stack/sheet/mineral/sandstone = 50,
```
```suggestion
		/obj/item/stack/sheet/mineral/sandstone/thirty = null,
```

ImogenOC commented on modular_nova/modules/admin_tech/code/admin-belts.dm:
```diff
@@ -0,0 +1,100 @@
+	new	/obj/item/rwd/admin(src)
+	new	/obj/item/analyzer/ranged(src)
+	new	/obj/item/construction/rtd/admin(src)
+	new	/obj/item/construction/rcd/arcd/mattermanipulator(src)
```
weewoo, i need to make the subspace version of this and I think this was a silly from copying stuff around

ImogenOC reviewed (commented):
batch review by zerg, thanks synthsnek

ImogenOC committed:
Apply suggestions from code review, for real

I dont use github ok

Co-authored-by: Zergspower <22140677+Zergspower@users.noreply.github.com>

ImogenOC committed:
revokes 3/4 chrono modules

was correctly reminded we have a moratorium on its usage

ImogenOC committed:
fixes a single missed revision

ImogenOC committed:
merges the map icons i'd kept removing because i thought i messed up

FlufflesTheDog commented on modular_nova/modules/admin_tech/code/admin_items.dm:
```diff
@@ -0,0 +1,276 @@
+		/obj/item/stack/sheet/hairlesshide = 50,
+		/obj/item/stack/sheet/hot_ice = 50,
+		/obj/item/stack/sheet/mineral/sandbags/fifty = null,
+		/obj/item/stack/sheet/cloth = 50,
```
Pretty sure this isn't going to work like this implies

The 5 isn't how many times to spawn this item, it's just an amount, which is passed to initialize, which overrides the typed stack size. So the type will be /cloth/ten, but it'll only have 5 cloths
(The nulls above work because null is the same as not passing any argument, so it defaults to the amount set by the type)

ImogenOC commented:
Stalkeros is looking at my boxcutter request

ImogenOC committed:
trying to fix icon merge conflicts

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into bluespace-tech-rework

ImogenOC committed:
fixes conflicts with updates done since I last touched this

also removes the central file I had.

cut scope significantly for this PR, field test first for further outfits

ImogenOC committed:
obligatory new map icons

ImogenOC committed:
all of today's work

okay lets see uhhh
custom modsuit sprites done
new custom mod plates
custom IDs done
custom remote done
minor cleanup of module and item lists
expands construction bag list

ImogenOC committed:
Merge branch 'NovaSector:master' into bluespace-tech-rework

ImogenOC committed:
subspace matter manipulator

ImogenOC committed:
subspace rapid lighting device

hey so did you ever stop and think about how a 'mini' version of something implies the existence of a normal sized version? yeah that struck me too. so I went and lo and behold, there is a normal sized rapid lighting device that is almost entirely unused. how? it's a tool that seems to only be mapped in or purchased via cargo, but who is doing that? the mini version is rcf printable, and has 100 matter compared to the normal version of... 200. why? syncit, gandalf, and three years ago, in the hand device refactor, apparently. the mini version isnt even smaller? its just less matter. i hate it here.

ImogenOC committed:
subspace emag & doorjack

why this one does not already exist is so far beyond my reasoning.

ImogenOC committed:
subspace light replacer

this one wasn't too funky.

ImogenOC committed:
subspace forcefield and atmos holofan projectors

ImogenOC committed:
fixes small stupid

ImogenOC committed:
self-powered tiny fan capsule

thanks for the idea, bubber.
contains a tiny fan, a hyper insul floor, and a noop area so it can be deployed anywhere

ImogenOC committed:
subspace plumbing tool and shears

the more of these I go through, the more horrified I am

ImogenOC committed:
the subspace shears

ImogenOC committed:
subspace tanks - atmos bullshittery

adds new internals tanks with huge capacity and some custom gas mixes

ImogenOC committed:
final adjustments for the night

ImogenOC commented:
I'm going to split the rest of the outfits off into it's own PR, as the ERTs that we currently have suck and aren't rebalanced to modern nova at all.

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
part one of the plushie spawners and code

i am getting semantic satiation from this and need to take a break

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into bluespace-tech-rework

ImogenOC committed:
nova plushies nova plushies nova plushies AHHH my brain

ImogenOC committed:
i forgot how inheritance worked for a minute there

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
subspace gloves: tackler + kaza rukl + combat gloves

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into bluespace-tech-rework

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
fixes the subspace boot's pockets and makes them phat. put a wylom in there or something.

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into bluespace-tech-rework

ImogenOC committed:
subspace rped bugs fixed

ImogenOC committed:
removes a todo, thats all for now

ImogenOC committed:
subspace contacts all fixed up!

ImogenOC committed:
hahahaha nice

ImogenOC committed:
code cleanups

ImogenOC committed:
fixes dme

ImogenOC committed:
someone finally explained to me how master files should be used

ImogenOC committed:
more icon corrections

ImogenOC committed:
Update shoes.dmi

ImogenOC committed:
icon cleanup, subspace gauntlet icons

ImogenOC committed:
wow theres a lot going on here

started code refactor while doing sprites, reorganized some stuff again... new items, new sprites, new names, stuff is out of date elsewhere, but we're moving in the right direction.
i also started calling them admin items instead of debug items because I feel like debug has different connotations than what I'm doing

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
continues refactor

ImogenOC committed:
somtimes is maybe good usually is shit

ImogenOC committed:
thanks again, eris, saving me from myself

ImogenOC committed:
more sprites, jackets, and the visor has phasing now

ImogenOC committed:
tonights work. i dont remember if this was compiling or not

ImogenOC committed:
some changes. mostly theory crafting in the background

ImogenOC committed:
Merge branch 'NovaSector:master' into bluespace-tech-rework

ImogenOC committed:
pipe dreams and pdas

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into bluespace-tech-rework

ImogenOC committed:
new laser pointer, fixes stupid elsewhere

i just be yapping in the comments, really

ImogenOC committed:
bunch of work today

the tech visors now reveal orbiting ghosts, and can summon ghosts to look at you
annnd added a bunch of reagent containers
annnd fixed bugs
annnd fixed pda. i think. comes with a meteor pen now
annnd made a bunch of admin reagent containers
annnd did some lore stuff
annnd prepped to make new adminordrazine

ImogenOC committed:
apparently this is how ill have to handle wires

that and skillchips

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
INDENTS???

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into bluespace-tech-rework

ImogenOC committed:
stuff and things. this is taking forever

ImogenOC committed:
wow thats a lot moon

ImogenOC commented:
alright, solution for the debug boxes is in, i just reused the bsrped spawning panel code and its pretty nice. got confirmation from goofball that theres no better solutions for the garbage stuff than what is already done ( atleast that he says). nothing much to be done there

ic spawning menu is now modernized in the deeper sections, should work on all options... custom needs more checking.

adds a new trait for seeing wires.

sets up some code for the slime pendant, which im calling the cytotheca, to give a place to shove all the slime extracts an admin needs. also starts prepping some stupid awful code to pass godmode through a ctrl shift click with something that isn't clothing.... so i have to go about this in a silly way... maybe i shouldve made the cytotheca into normal clothing and dealt with those issues separately... uaaaahhhhh

ImogenOC committed:
more work

ImogenOC committed:
more work

ImogenOC committed:
notes

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
lots more work done

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
more fixes

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into admin-tech-module

ImogenOC committed:
code cleanup, update readme.md

ImogenOC committed:
cleanup and fixes

ImogenOC committed:
more touchups

ImogenOC committed:
me when i dont use a define

Neocloudy committed:
Documentation changes, and trying to split these files up

Neocloudy committed:
nope

Neocloudy committed:
Merge branch 'bluespace-tech-rework' of https://github.com/imogenoc/mewnnovafork into imogenoc-6628-patch

Neocloudy committed:
CLIENT_FROM_VAR

ImogenOC committed:
icon state fix for the carbine, fixes watering can vol

Neocloudy committed:
Compiles, some more documentation made to use autodoc format, also some optimizations

Neocloudy committed:
I changed my mind about those optimizations

Neocloudy committed:
The thing this comment says is needed isn't actually needed

Neocloudy committed:
This can also go

Neocloudy committed:
redundant comment

Neocloudy committed:
Merge branch 'bluespace-tech-rework' of https://github.com/imogenoc/mewnnovafork into imogenoc-6628-patch

Neocloudy committed:
Format

Neocloudy committed:
Because of skew/conflicts, these don't need to be split up yet. Also trailing commas

Neocloudy committed:
Storage fixes?

Neocloudy committed:
Comments. Also this type doesn't change anything yet so it can lose the variable lines

Neocloudy committed:
A

Neocloudy committed:
?

ImogenOC committed:
Merge pull request #2 from Neocloudy/imogenoc-6628-patch

Mainly code cleanup-oriented changes

ImogenOC committed:
moves code into a new pr

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into admin-tech-module

ImogenOC committed:
MORE BULLSHIT.

ImogenOC committed:
Merge branch 'master' into bluespace-tech-rework

ImogenOC committed:
robo bases

ImogenOC committed:
pestering with this shit some more but im bad

vinylspiders commented on modular_nova/modules/admin_tech/code/debug_boxes.dm:
```diff
@@ -0,0 +1,381 @@
+
+// Fun Boxes and Spawners//
+// Nova Plushie Spawners, no filtering
+/obj/effect/spawner/random/entertainment/plushie/nova/
```
```suggestion
/obj/effect/spawner/random/entertainment/plushie/nova
```

vinylspiders commented on modular_nova/modules/admin_tech/code/debug_boxes.dm:
```diff
@@ -0,0 +1,381 @@
+/obj/effect/spawner/random/entertainment/plushie/nova/donator/all/
+	name = "nova sector donator all plushies spawner"
+	icon_state = "plushie"
+	spawn_all_loot = TRUE
```
This is a dupe of the one above

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_belts.dm:
```diff
@@ -0,0 +1,82 @@
+/obj/item/storage/belt/utility/admin/full/bluespace
+	name = "\improper bluespace technician's satchel"
+	desc = "A hand crafted satchel made of the finest materials, processes, and a creatively integrated bluespace anomaly core. \
+	This one was carefully prestocked and organized by a Central Command Quartermaster to service an engineering aligned technician."
```
```suggestion
		This one was carefully prestocked and organized by a Central Command Quartermaster to service an engineering aligned technician."
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_belts.dm:
```diff
@@ -0,0 +1,82 @@
+/obj/item/storage/belt/utility/admin/full/subspace
+	name = "\improper subspace technician's satchel"
+	desc = "A hand crafted satchel made of the finest materials, processes, and a creatively integrated bluespace anomaly core. \
+	This one was carefully prestocked and organized by a Central Command Quartermaster to offer an extreme selection of goodies."
```
```suggestion
		This one was carefully prestocked and organized by a Central Command Quartermaster to offer an extreme selection of goodies."
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_clothing.dm:
```diff
@@ -0,0 +1,532 @@
+	var/xray = FALSE
+	pickup_sound = SFX_GOGGLES_PICKUP
+	drop_sound = SFX_GOGGLES_DROP
+	equip_sound = SFX_GOGGLES_EQUIP
```
```suggestion
	pickup_sound = SFX_GOGGLES_PICKUP
	drop_sound = SFX_GOGGLES_DROP
	equip_sound = SFX_GOGGLES_EQUIP
	var/xray = FALSE
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_clothing.dm:
```diff
@@ -0,0 +1,532 @@
+		/datum/gas/proto_nitrate,
+		/datum/gas/tritium,
+		/datum/gas/zauker,
+		)
```
```suggestion
	)
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_clothing.dm:
```diff
@@ -0,0 +1,532 @@
+	if(admin_godmode)
+		ADD_TRAIT(user, TRAIT_GODMODE, REF(user))
+//	if(user & slot & slot_flags & admin_godmode)
+		ADD_TRAIT(user, TRAIT_GODMODE, REF(user))
```
```suggestion
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_clothing.dm:
```diff
@@ -0,0 +1,532 @@
+/obj/item/clothing/under/admin
+	name = "bluespace techsuit"
+	desc = "A perfectly tailored and customized skin suit made specifically for this technician. \
+	Composed of experimental textiles, and assembled with the legendary Bluespace Sewing Machine, it fits the body with perfect comfort, and carries an air of security."
```
```suggestion
		Composed of experimental textiles, and assembled with the legendary Bluespace Sewing Machine, it fits the body with perfect comfort, and carries an air of security."
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_clothing.dm:
```diff
@@ -0,0 +1,532 @@
+		balloon_alert(user, "must be worn!")
+		return CLICK_ACTION_BLOCKING
+
+	admin_spacewalk = !admin_spacewalk
+	if(admin_spacewalk)
+		attach_clothing_traits(TRAIT_SPACEWALK)
+	else
+		detach_clothing_traits(TRAIT_SPACEWALK)
```
This does the same thing in both branches

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_items.dm:
```diff
@@ -0,0 +1,1265 @@
+/obj/item/storage/bag/construction/admin
+	name = "bluespace construction bag"
+	desc = "An artisinally crafted pocket liner utilizing advanced technologies, techniques, and materials. \
+	Peeking inside the pocket, cherenkov-esque radiation illuminates a mass of materials and supplies."
```
```suggestion
		Peeking inside the pocket, cherenkov-esque radiation illuminates a mass of materials and supplies."
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_items.dm:
```diff
@@ -0,0 +1,1265 @@
+/obj/item/gun/energy/cell_loaded/medigun/admin
+	name = "subspace medigun"
+	desc = "VeyMed was not happy with this one, but they didn't get much of a say in it's manufacture. This 'aftermarket' (still manufactured by VeyMed) specification comes loaded with every cell. \
+	Test users said the switching was 'cumbersome' and that a 'floating radial' was a cooler choice, but the acquisitions manager lacked ability to describe the design to the producer."
```
```suggestion
		Test users said the switching was 'cumbersome' and that a 'floating radial' was a cooler choice, but the acquisitions manager lacked ability to describe the design to the producer."
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_items.dm:
```diff
@@ -0,0 +1,1265 @@
+/obj/item/laser_pointer/admin
+	name = "subspace laser pointer"
+	desc = "It's a fidget toy with a warning label, describing why you should definitely avoid pointing this rapidly enough for the universe to 'ratelimit' you, whatever that means. \
+	Turning it over, you notice a crudely hand-etched representation of a crying cyborg."
```
```suggestion
		Turning it over, you notice a crudely hand-etched representation of a crying cyborg."
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_items.dm:
```diff
@@ -0,0 +1,1265 @@
+			/datum/laser_weapon_mode/admin/meteor,
+			/datum/laser_weapon_mode/admin/ion,
+			/datum/laser_weapon_mode/admin/plasmacutter,
+			/datum/laser_weapon_mode/admin/gravity
```
```suggestion
			/datum/laser_weapon_mode/admin/gravity,
```

vinylspiders commented on modular_nova/modules/admin_tech/code/admin_robots.dm:
```diff
@@ -0,0 +1,151 @@
+		/obj/item/holosign_creator/cyborg,
+		/obj/item/stamp/chameleon,
+		/obj/item/borg_shapeshifter,
+		)
```
```suggestion
	)
```

ImogenOC committed:
Apply suggestions from code review

Co-authored-by: Bloop <13398309+vinylspiders@users.noreply.github.com>

ImogenOC committed:
A FEW LINES THANKS GIT

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into admin-tech-module

ImogenOC committed:
Admin borgs now work in a basic sense

ImogenOC committed:
Central Command manufacturing company make

ImogenOC committed:
sets up ITEM_SLOT_ADMIN

ImogenOC committed:
like, a bunch of shit really

ImogenOC committed:
technician stealth trait, untested rn

ImogenOC committed:
COOL SHIT IN HERE SO COOL SO COOL

ImogenOC committed:
some more cleanup

ImogenOC committed:
silent spawn addition

ImogenOC committed:
More fixes across the entire pr.

ImogenOC committed:
Update modular_nova/modules/admin_tech/code/admin_clothing.dm

Co-authored-by: Bloop <13398309+vinylspiders@users.noreply.github.com>

ImogenOC committed:
Update modular_nova/modules/admin_tech/code/admin_items.dm

Co-authored-by: Bloop <13398309+vinylspiders@users.noreply.github.com>

ImogenOC committed:
Update modular_nova/modules/admin_tech/code/admin_items.dm

Co-authored-by: Bloop <13398309+vinylspiders@users.noreply.github.com>

ImogenOC committed:
Update modular_nova/modules/admin_tech/code/admin_items.dm

Co-authored-by: Bloop <13398309+vinylspiders@users.noreply.github.com>

ImogenOC committed:
Update modular_nova/modules/admin_tech/code/admin_items.dm

Co-authored-by: Bloop <13398309+vinylspiders@users.noreply.github.com>

ImogenOC committed:
Starting code cleanup

ImogenOC committed:
Merge branch 'bluespace-tech-rework' of https://github.com/ImogenOC/MewnNovaFork into admin-tech-module

ImogenOC committed:
More code reviewing, comments, cleanup...

ImogenOC committed:
Update admin_robots.dm

ImogenOC committed:
Update readme.md

ImogenOC committed:
me when i hotkeys things

ImogenOC committed:
outfit and obj cleanup and integration

ImogenOC committed:
Update readme.md

ImogenOC committed:
first major playtesting bugfixes and cleanup

ImogenOC committed:
new element, more comments

ImogenOC committed:
I CANT STOP TOUCHING IT, KILL ME

ImogenOC committed:
i cant stop i cant stop ik hept honger

ImogenOC committed:
mal has given me a pr review guide and i gotta copa

ImogenOC committed:
lots of holder checks and code review

ImogenOC committed:
unfucks something I did

ImogenOC committed:
Update admin_items.dm

ImogenOC committed:
thank you sammy for the recommendations

ImogenOC committed:
bitflags and org

ImogenOC committed:
code review from mac

Co-Authored-By: Happyowl93 <231794227+Happyowl93@users.noreply.github.com>

ImogenOC committed:
merge conflict fix

ImogenOC committed:
runtimes and fixes

carbine runtime is smushed
implements a strategy on the modular laser rifle to accept different manufacturers, used it for mine

thanks lemons

FlufflesTheDog commented on code/__DEFINES/~nova_defines/obj_flags.dm:
```diff
@@ -13,3 +13,11 @@
+/// Prevents an item from being stripped unless you are an admin privileges holder.
+#define NOSTRIP (1<<50)
+/// Is this an Admin item? Adds an 'admistrative' obj detail, similar to how sec gloves show they cuff quickly or insuls provide insulation
+#define ADMIN_ITEM (1<<51)
```
```suggestion
/// Prevents an item from being stripped unless you are an admin privileges holder.
#define NOSTRIP (1<<3)
/// Is this an Admin item? Adds an 'admistrative' obj detail, similar to how sec gloves show they cuff quickly or insuls provide insulation
#define ADMIN_ITEM (1<<4)
```
This is way too large. Byond uses 32 bit floating point numbers, which can only accurately represent integer values of up to 24 bits. 1 << 23 is the highest a bitflag should ever go.

ImogenOC committed:
Update code/__DEFINES/~nova_defines/obj_flags.dm

Per senior maint correction over assistant maint correction.

Co-authored-by: FlufflesTheDog <piecopresident@gmail.com>
